File: Algorithm/CumulativeSum/CumulativeSum.ts
import * as fs from "fs";

function solveFromInput(input: string): string {
	const lines = input.trim().split("\n");
	const [N, Q] = lines[0].split(" ").map(Number);
	const A = lines[1].split(" ").map(Number);
	const queries = lines.slice(2).map((line) => line.split(" ").map(Number));

	// 累積和を計算
	const S: number[] = Array(N + 1).fill(0);
	for (let i = 1; i <= N; i++) {
		S[i] = S[i - 1] + A[i - 1];
	}

	// 各質問に答える
	const results: number[] = [];
	for (const [L, R] of queries) {
		results.push(S[R] - S[L - 1]);
	}

	// 結果を改行区切りで返す
	return results.join("\n");
}

// 標準入力からデータを読み込む
const input = fs.readFileSync("/dev/stdin", "utf8");
const output = solveFromInput(input);
console.log(output);

File: Algorithm/CumulativeSum/CumulativeSum.py
def main():
    import sys
    input = sys.stdin.read
    data = input().splitlines()

    # N と Q を取得
    N, Q = map(int, data[0].split())

    # 来場者数データを取得
    A = list(map(int, data[1].split()))

    # 累積和を計算
    prefix_sum = [0] * (N + 1)
    for i in range(1, N + 1):
        prefix_sum[i] = prefix_sum[i - 1] + A[i - 1]

    # 質問を処理
    results = []
    for i in range(Q):
        L, R = map(int, data[2 + i].split())
        # L 日目から R 日目までの合計を計算
        results.append(prefix_sum[R] - prefix_sum[L - 1])

    # 結果を出力
    print("\n".join(map(str, results)))

if __name__ == "__main__":
    main()
